src/settingsdialog.cpp

```diff
@@ -290,3 +297,128 @@ void SettingsDialog::addCallgraphPage()
                                      callgraphPage->functionColor->color().name());
     });
 }
+
+void SettingsDialog::addSshPage()
+{
+    const auto sshPath = QStandardPaths::findExecutable(QLatin1String("ssh"));
+
+    // only show page if ssh is installed
+    if (sshPath.isEmpty()) {
+        return;
+    }
+
+    auto page = new QWidget(this);
+    auto item = addPage(page, tr("SSH"));
+    item->setHeader(tr("SSH Settings"));
+    item->setIcon(QIcon::fromTheme(QStringLiteral("preferences-system-windows-behavior")));
+
+    sshPage->setupUi(page);
+
+    auto deviceConfig = config()->group("devices");
+
+    auto saveFunction = [this](KConfigGroup group) {
+        group.writeEntry("hostname", sshPage->ipLineEdit->text());
+        group.writeEntry("username", sshPage->usernameLineEdit->text());
+        group.writeEntry("sshoptions", sshPage->sshOptionsLineEdit->text());
+    };
+
+    auto restoreFunction = [this](const KConfigGroup& group) {
+        const auto hostname = group.readEntry("hostname");
+        const auto username = group.readEntry("username");
+        const auto sshOptions = group.readEntry("sshoptions");
+
+        sshPage->ipLineEdit->setText(hostname);
+        sshPage->usernameLineEdit->setText(username);
+        sshPage->sshOptionsLineEdit->setText(sshOptions);
+    };
+
+    MultiConfigWidget* devices = new MultiConfigWidget(this);
+
+    // TODO: fix tab order
+
+    connect(devices, &MultiConfigWidget::saveConfig, this, saveFunction);
+    connect(devices, &MultiConfigWidget::restoreConfig, this, restoreFunction);
+
+    devices->setConfig(deviceConfig);
+    devices->restoreCurrent();
+
+    sshPage->groupBox->layout()->replaceWidget(sshPage->multiConfigWidget, devices);
+
+    Settings* settings = Settings::instance();
+    sshPage->sshaskpassLineEdit->setText(settings->sshaskpassPath());
+
+    connect(settings, &Settings::sshaskpassChanged, this,
+            [this](const QString& path) { sshPage->sshaskpassLineEdit->setText(path); });
+
+    connect(sshPage->testButton, &QPushButton::pressed, this, [this, devices] {
+        auto ssh = new QProcess(this);
+        ssh->setProgram(QStandardPaths::findExecutable(QLatin1String("ssh")));
+
+        const auto hostname =
```

Comment:
it should be allowed to not specify a user, in which case ssh will use the default internally (either current or whatever is provided in `~/.ssh/.config)`. but I believe connecting to `@localhost` e.g. is going to fail, the `@` prefix must not be added in such a case